### DISCUSSION TITLE
Issue where logfile is not created when running dmalloc.

### DISCUSSION THREAD

**Discussion by YJHcc** [j256/dmalloc#115] [Q&A] [answered]
* User Information : 
ubuntu ver : 20.04
gcc ver : gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
target : jetson board (aarch64 architecture)
system info. : 5.10.104-tegra

hello.
I have a question regarding the use of dmalloc.

The desired task is to compile the test.c source code and check memory usage(or memory leakage) using dmalloc. At this time, the dmalloc-related git clone process I applied is explained as follows.
  - git clone https://github.com/j256/dmalloc.git
  - ./configure
  - make && make install
  
* dmalloc settings
  - dmalloc -l ./logfile -i 100 high > dmalloc_setup (or dmalloc -l /home/jetson/profile_test1/dmalloc/logfile -i 100 high > dmalloc_setup)
  - source dmalloc_setup
  - echo $DMALLOC_OPTIONS
  
* run
  - gcc -g -DDMALLOC -o test test.c -ldmalloc
  - ./test
  - cat logfile

The code in my test.c is as follows:
```
#include <stdio.h>
#include <stdlib.h>
#ifdef DMALLOC
#include <dmalloc.h>
#endif

int main() {
   char *ch_p = (char *)malloc(10);

  for (int j = 0; j < 2; j++)
      malloc(100);            // no free

  calloc(16, 16);             // no free
  return 0;
}
```

But, at this time, the logfile is not created. Any problem?

**Comment by j256** [comment, marked as answer]
Huh this works for me.  Some things to try.  First, after you source the settings, you should be able to run `dmalloc` again to make sure that the high debug flags and the logpath are set:

    $ ./dmalloc
    Debug-Flags`` 0x4f4ed03 (83160323) (internal token: high)
    ...
    Logpath      './logfile'

Or print out the DMALLOC env variable:

    $ env | grep DMALLOC
    DMALLOC_OPTIONS=debug=0x4f4ed03,inter=100,log=./logfile

Next, do a `nm test` on your utility to make sure that it linked with libdmalloc.a correctly.  You should see a bunch of _dmalloc symbols:

    $ nm test | grep dmalloc
    00000000000f11c0 B _dmalloc_aborting_b
    00000000000f11f0 B _dmalloc_address
    ...

Lastly, make sure you are running your `test` and not the shell built in.  Maybe use `./test`.

    $ rm -f logfile
    $ ./test
    $ ls -l logfile
    -rw-r----- 1 j256 staff 2114 Jun 21 18:04 logfile

**Reply by matthewluckie** [reply]
I've been trying to use dmalloc on a debian trixie system.  I first tried a docker container and then a virtual machine.  Unfortunately, I'm not getting a logfile.

Using the same test.c and testing procedure as above on an ubuntu 24.04 system works.  On both systems, I installed dmalloc (5.5.2) from apt.

```
$ dmalloc
Debug-Flags 0x4e48503 (82085123) (internal token: low)
Address      not-set
Interval     10000000000
Lock-On      not-set
Logpath      'logfile'
Mem-Limit    not-set
Start        not-set

Debug Malloc Utility: http://dmalloc.com/
  For a list of the command-line options enter: dmalloc --usage
$ ldd testx
        linux-vdso.so.1 (0x00007f5e7ad4d000)
        libdmalloc.so.5 => /lib/x86_64-linux-gnu/libdmalloc.so.5 (0x00007f5e7ac49000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f5e7aa53000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f5e7ad4f000)
$ nm testx | grep dmalloc
                 U dmalloc_malloc
$ ./testx 
$ ls -l
-rw-rw-r--  1 mjl mjl   260 Aug 18 13:31 test.c
-rwxrwxr-x  1 mjl mjl 17344 Aug 18 13:31 testx
```
